core/src/test/java/dk/alexandra/fresco/logging/TestGenericLoggingDecorators.java

```diff
@@ -0,0 +1,301 @@
+      Assert.assertThat(loggedValues.get(NetworkLoggingDecorator.NETWORK_TOTAL_BATCHES), Is.is(0));
+      Assert.assertThat(loggedValues.get(NetworkLoggingDecorator.NETWORK_MAX_BYTES), Is.is(0));
+      Assert.assertThat(loggedValues.get(NetworkLoggingDecorator.NETWORK_MIN_BYTES), Is.is(Integer.MAX_VALUE));
+      Assert.assertTrue(((Map<Integer, Integer>)loggedValues.get(NetworkLoggingDecorator.NETWORK_PARTY_BYTES)).isEmpty());
+    }
+  }
+  
+  @Test
+  public void testBatchEvaluationLoggingDecorator() throws Exception {
+    
+    TestThreadRunner.TestThreadFactory<DummyArithmeticResourcePool, ProtocolBuilderNumeric> f
+        = new SqrtTests.TestSquareRoot<>();
+
+    EvaluationStrategy evalStrategy = EvaluationStrategy.SEQUENTIAL; 
+
+    Map<Integer, NetworkConfiguration> netConf = getNetConf();
+
+    Map<Integer, TestThreadRunner.TestThreadConfiguration<DummyArithmeticResourcePool, ProtocolBuilderNumeric>> conf =
+        new HashMap<>();
+
+    Map<Integer, ListLogger> pls = new HashMap<>();
+    List<PerformanceLogger> decoratedLoggers = new ArrayList<>();
+    for (int playerId : netConf.keySet()) {
+      pls.put(playerId, new ListLogger());
+      NetworkConfiguration partyNetConf = netConf.get(playerId);
+      
+      DummyArithmeticProtocolSuite ps = new DummyArithmeticProtocolSuite(mod, 200);
+      BatchEvaluationStrategy<DummyArithmeticResourcePool> strat = evalStrategy.getStrategy();
+      BatchEvaluationLoggingDecorator<DummyArithmeticResourcePool> decoratedStrat =
+          new BatchEvaluationLoggingDecorator<>(strat);
+      decoratedLoggers.add(decoratedStrat);
+      
+      ProtocolEvaluator<DummyArithmeticResourcePool, ProtocolBuilderNumeric> evaluator 
+          = new BatchedProtocolEvaluator<>(decoratedStrat, ps);
+      SecureComputationEngine<DummyArithmeticResourcePool, ProtocolBuilderNumeric> sce 
+          = new SecureComputationEngineImpl<>(ps, evaluator);
+      
+      Drbg drbg = new HmacDrbg();
+      TestThreadRunner.TestThreadConfiguration<DummyArithmeticResourcePool, ProtocolBuilderNumeric> ttc =
+          new TestThreadRunner.TestThreadConfiguration<>(sce,
+              () -> new DummyArithmeticResourcePoolImpl(playerId,
+                  netConf.keySet().size(), drbg, mod),
+              () -> {
+                return new KryoNetNetwork(partyNetConf);
+              });
+      conf.put(playerId, ttc);
+    }
+    TestThreadRunner.run(f, conf);
+
+    for (Integer pId : pls.keySet()) {
+      PerformanceLogger performanceLogger = decoratedLoggers.get(0);
+      performanceLogger.printToLog(pls.get(pId), pId);
+      String expectedOutput = performanceLogger.makeLogString(pId);
+      Assert.assertThat(pls.get(pId).getData().get(0), Is.is(expectedOutput));
+
+      Map<String, Object> loggedValues = performanceLogger.getLoggedValues(pId);
+      Assert.assertThat(loggedValues.get(BatchEvaluationLoggingDecorator.BATCH_COUNTER), Is.is(3496361));
```

Comment:
Again, this number is not really obvious at all. This goes for many of these tests.